Index: kss/core/configure.zcml
===================================================================
--- kss/core/configure.zcml	(revision 52771)
+++ kss/core/configure.zcml	(working copy)
@@ -3,7 +3,9 @@
            xmlns:zcml="http://namespaces.zope.org/zcml"
            xmlns:five="http://namespaces.zope.org/five">
 
-    <include file="permission.zcml"/>
+    <include package="zope.app.component" file="meta.zcml"/>
+    <include package="zope.app.security" file="meta.zcml"/> 
+    <include package="zope.app.security"/> 
 
  
     <!-- This include MUST come first! -->
